### ISSUE TITLE
GUI Advanced installation: unable to find partitions to install CLR_BOOT, CLR_SWAP, CLR_ROOT

### ISSUE THREAD

**Issue by stokito** [clearlinux/clr-installer#553]
I runned GParted and created all needed partitions but when I started the GUI installer again it still shows me "Missing CLR_BOOT partition label, Missing CLR_SWAP partition label, Missing CLR_ROOT partition label". Even when I pressed the Rescan media button the error persists.

But when I runned TUI installer in terminal `clr-installer` it was able to find this partitions and installation was successful.
See screenshots for details.
![GUI](https://user-images.githubusercontent.com/415502/65824266-89974300-e255-11e9-947e-641a08f620c4.png)

![TUI](https://user-images.githubusercontent.com/415502/65824264-89974300-e255-11e9-9cb6-7a9e54d25598.png)

![GParted](https://user-images.githubusercontent.com/415502/65824265-89974300-e255-11e9-9744-a00417d6338c.png)

**Environment (please complete the following information):**
 - Clear Linux OS Version: 31090
 - Installer Version 2.2.0

**Comment by mdhorn** [comment]
@stokito Were any of the partitions from /dev/sda mounted on the running OS? If so, we will mask that device as a possible install target to prevent modifying media which is in use.

The default ISO image should be disabling automount of media for this reason.

**Comment by mdhorn** [comment]
If you have the /root/clr-installer.log it may provide some insight as to why it was not seen.

**Comment by stokito** [comment]
[clr-installer.log](https://github.com/clearlinux/clr-installer/files/3680223/clr-installer.log)
From what I see the installer failed because of the `/dev/loop0` device.
I executed the command  `lsblk --exclude 1,2,11 -J -b -O`:
```json
{
   "blockdevices": [
      {"name":"loop0", "kname":"loop0", "path":"/dev/loop0", "maj:min":"7:0", "fsavail":null, "fssize":null, "fstype":"squashfs", "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":null, "ptuuid":null, "pttype":null, "parttype":null, "partlabel":null, "partuuid":null, "partflags":null, "ra":128, "ro":true, "rm":false, "hotplug":false, "model":null, "serial":null, "size":2493706240, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":true, "sched":"mq-deadline", "rq-size":256, "type":"loop", "disc-aln":0, "disc-gran":512, "disc-max":4294966784, "disc-zero":false, "wsame":0, "wwn":null, "rand":false, "pkname":null, "hctl":null, "tran":null, "subsystems":"block", "rev":null, "vendor":null, "zoned":"none"},
      {"name":"sda", "kname":"sda", "path":"/dev/sda", "maj:min":"8:0", "fsavail":null, "fssize":null, "fstype":null, "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":null, "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":null, "partlabel":null, "partuuid":null, "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":"Patriot_FLARE", "serial":"ACE90793176E00179487", "size":120034123776, "state":"running", "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"disk", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":null, "hctl":"2:0:0:0", "tran":"sata", "subsystems":"block:scsi:pci", "rev":"21.2", "vendor":"ATA     ", "zoned":"none",
         "children": [
            {"name":"sda1", "kname":"sda1", "path":"/dev/sda1", "maj:min":"8:1", "fsavail":null, "fssize":null, "fstype":"ntfs", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"Recover", "uuid":"4236826C36826131", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"de94bba4-06d1-4d40-a16a-bfd50179d6ac", "partlabel":"Basic data partition", "partuuid":"8de11847-0822-41b9-a09c-fad8d481440b", "partflags":"0x1", "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":471859200, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda2", "kname":"sda2", "path":"/dev/sda2", "maj:min":"8:2", "fsavail":null, "fssize":null, "fstype":"vfat", "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":"8A82-9A48", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"c12a7328-f81f-11d2-ba4b-00a0c93ec93b", "partlabel":"EFI system partition", "partuuid":"c5dea274-132d-484c-9411-2ac408ea49e7", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":103809024, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda3", "kname":"sda3", "path":"/dev/sda3", "maj:min":"8:3", "fsavail":null, "fssize":null, "fstype":null, "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":null, "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"e3c9e316-0b5c-4db8-817d-f92df00215ae", "partlabel":"Microsoft reserved partition", "partuuid":"e48aff89-2ffb-4736-ada7-4fd36ba92908", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":16777216, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda4", "kname":"sda4", "path":"/dev/sda4", "maj:min":"8:4", "fsavail":null, "fssize":null, "fstype":"ntfs", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"c", "uuid":"1E3483D63483AF79", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"ebd0a0a2-b9e5-4433-87c0-68b6b72699c7", "partlabel":"c", "partuuid":"2fa11279-9c9c-480a-88da-e16dd4c0a77a", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":49126834176, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda5", "kname":"sda5", "path":"/dev/sda5", "maj:min":"8:5", "fsavail":null, "fssize":null, "fstype":"ntfs", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"WinRecovery", "uuid":"8454B19954B18E84", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"de94bba4-06d1-4d40-a16a-bfd50179d6ac", "partlabel":null, "partuuid":"62f6b99a-1e96-4ad6-bbd9-6830f16f5d04", "partflags":"0x1", "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":881852416, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda6", "kname":"sda6", "path":"/dev/sda6", "maj:min":"8:6", "fsavail":null, "fssize":null, "fstype":"crypto_LUKS", "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":"df448e68-a776-4a8b-9edc-5557e0b22a29", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"ca7d7ccb-63ed-4c53-861c-1742536059cc", "partlabel":null, "partuuid":"c415536b-cfea-47c8-8917-9e95e3aefdb7", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":4979687424, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda7", "kname":"sda7", "path":"/dev/sda7", "maj:min":"8:7", "fsavail":null, "fssize":null, "fstype":null, "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":null, "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"21686148-6449-6e6f-744e-656564454649", "partlabel":null, "partuuid":"2987469a-2dcf-43e2-80af-fb2f888a1da7", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":1048576, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda8", "kname":"sda8", "path":"/dev/sda8", "maj:min":"8:8", "fsavail":null, "fssize":null, "fstype":"ext4", "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":"94498bf5-59bb-4797-959b-12b27c678220", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"0fc63daf-8483-4772-8e79-3d69d8477de4", "partlabel":"ubuntu", "partuuid":"9cd4c5f2-22ee-4b2f-a20e-cd32f3ee7714", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":29439819776, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda9", "kname":"sda9", "path":"/dev/sda9", "maj:min":"8:9", "fsavail":null, "fssize":null, "fstype":"ext4", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"workspace", "uuid":"b14b359d-31e2-467d-a13f-15aeb736bd52", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"0fc63daf-8483-4772-8e79-3d69d8477de4", "partlabel":null, "partuuid":"c37ee72e-edaa-4da2-9b11-a0cdd4a1cbbe", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":9999220736, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda10", "kname":"sda10", "path":"/dev/sda10", "maj:min":"8:10", "fsavail":null, "fssize":null, "fstype":"crypto_LUKS", "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":"398e8283-741d-4a8e-b7f8-630614d77dd6", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"ca7d7ccb-63ed-4c53-861c-1742536059cc", "partlabel":null, "partuuid":"23dc66fd-a927-43cf-ad13-88530aee4738", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":5012193280, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda11", "kname":"sda11", "path":"/dev/sda11", "maj:min":"8:11", "fsavail":null, "fssize":null, "fstype":"vfat", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"BOOT", "uuid":"48E9-F4D4", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"c12a7328-f81f-11d2-ba4b-00a0c93ec93b", "partlabel":"CLR_BOOT", "partuuid":"e9b471b6-0c9b-4c54-b510-91fed998f1cf", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":157286400, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda12", "kname":"sda12", "path":"/dev/sda12", "maj:min":"8:12", "fsavail":null, "fssize":null, "fstype":"swap", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"swap", "uuid":"76df8038-e61c-4fde-8dad-d73aacf63ffc", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"0657fd6d-a4ab-43c4-84e5-0933c84b4f4f", "partlabel":"CLR_SWAP", "partuuid":"32dff88b-c11d-48c5-bbf7-37fc13ace039", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":268435456, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"},
            {"name":"sda13", "kname":"sda13", "path":"/dev/sda13", "maj:min":"8:13", "fsavail":null, "fssize":null, "fstype":"ext4", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"root", "uuid":"71c501c0-e7dc-405e-9bef-cf0c410060ab", "ptuuid":"b9687a7d-6d6c-43a9-8ff9-d17901dcfdd9", "pttype":"gpt", "parttype":"4f68bce3-e8cd-4db1-96e7-fbcaf984b709", "partlabel":"CLR_ROOT", "partuuid":"4070017e-cdd4-454b-b48c-955ef6bc1f79", "partflags":null, "ra":1024, "ro":false, "rm":false, "hotplug":false, "model":null, "serial":null, "size":19573768192, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":false, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":512, "disc-max":2147450880, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sda", "hctl":null, "tran":null, "subsystems":"block:scsi:pci", "rev":null, "vendor":null, "zoned":"none"}
         ]
      },
      {"name":"sdb", "kname":"sdb", "path":"/dev/sdb", "maj:min":"8:16", "fsavail":null, "fssize":null, "fstype":null, "fsused":null, "fsuse%":null, "mountpoint":null, "label":null, "uuid":null, "ptuuid":"6e7fde40", "pttype":"dos", "parttype":null, "partlabel":null, "partuuid":null, "partflags":null, "ra":1024, "ro":false, "rm":true, "hotplug":true, "model":"Transcend_8GB", "serial":"PC8AG5VS", "size":7813988352, "state":"running", "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":true, "sched":"bfq", "rq-size":1024, "type":"disk", "disc-aln":0, "disc-gran":0, "disc-max":0, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":null, "hctl":"6:0:0:0", "tran":"usb", "subsystems":"block:scsi:usb:pci", "rev":"8.07", "vendor":"JetFlash", "zoned":"none",
         "children": [
            {"name":"sdb1", "kname":"sdb1", "path":"/dev/sdb1", "maj:min":"8:17", "fsavail":null, "fssize":null, "fstype":"vfat", "fsused":null, "fsuse%":null, "mountpoint":null, "label":"CLR_ISO", "uuid":"2DEA-F0D0", "ptuuid":"6e7fde40", "pttype":"dos", "parttype":"0xc", "partlabel":null, "partuuid":"6e7fde40-01", "partflags":"0x80", "ra":1024, "ro":false, "rm":true, "hotplug":true, "model":null, "serial":null, "size":7813857280, "state":null, "owner":null, "group":null, "mode":"brw-rw----", "alignment":0, "min-io":512, "opt-io":0, "phy-sec":512, "log-sec":512, "rota":true, "sched":"bfq", "rq-size":1024, "type":"part", "disc-aln":0, "disc-gran":0, "disc-max":0, "disc-zero":false, "wsame":0, "wwn":null, "rand":true, "pkname":"sdb", "hctl":null, "tran":null, "subsystems":"block:scsi:usb:pci", "rev":null, "vendor":null, "zoned":"none"}
         ]
      }
   ]
}
```

**Comment by jwakre** [comment]
From the attached log: `CLR_ROOT must be >= 21.5G `

It looks like CLR_ROOT does not meet the minimum space requirement. The error message on the GUI could be improved.

**Comment by stokito** [comment]
Interesting why TUI was ok with that

**Comment by mdhorn** [comment]
@stokito  The TUI works due to a poor assumption we've been making about disk requirements; the text-based installer was primarily for server installs (~8GB required) and the graphical for desktop installs (~20GB required).
Unfortunately, though swupd is excellent about minimizing the amount of data to down, it is not currently capable of telling us how much space is needed for a given set of bundles -- so we have estimates for the two use cases.

This has made us think of a better way of determining if we are a server .vs. desktop install, and we will add that in the future.

**Comment by mdhorn** [comment]
Fixed in clr-installer release [2.3.1](https://github.com/clearlinux/clr-installer/releases/tag/2.3.1) in Clear Linux OS release 31270 or later.
